### PR TITLE
Help find and troubleshoot pprof installation

### PR DESCRIPTION
**Pull request by wlandau**
# Summary

To address #13 and #14, this PR exports 3 functions:

- `pprof_path()`: return the path to `pprof` if it exists and `""` otherwise.
- `assert_pprof()`: error if `pprof` is not found.
- `test_pprof()`: tests the `pprof()` function on a small example to make sure the `pprof` installation is usable.

By default, each function prints console messages whenever it struggles to find `pprof`. Set `verbose = FALSE` to turn these messages off.

# Related GitHub issues and pull requests

- Ref: #13, #14

# Checklist

- [x] I understand and agree to `proffer`'s [code of conduct](https://github.com/r-prof/proffer/blob/master/CODE_OF_CONDUCT.md).
- [x] I have listed any substantial changes in the [development news](https://github.com/r-prof/proffer/blob/master/NEWS.md).
- [x] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/r-prof/proffer/tree/master/tests/testthat) for any new functionality. Feel free to use `skip("interactive only")` for any tests that cannot be fully automated.
- [x] This pull request is not a [draft](https://github.blog/2019-02-14-introducing-draft-pull-requests).

### CODE REVIEW

**Comment by wlandau** [comment]
`pprof_path()` still does not work on Windows. I need to investigate when I get back to a Windows machine.

**Comment by wlandau** [comment]
As of https://github.com/r-prof/proffer/pull/15/commits/f76bde56796396e83fee00f94430c94974f18303, `proffer` searches for `pprof` on Windows too (if `pprof_path` is not set). So I think this PR completely solves #14. Maybe #13 as well, unless we want an all-in-one check for future system dependencies we may add in addition to `pprof`.

**Comment by wlandau** [comment]
@krlmlr, what do you think about these new functions for `proffer`?

File: README.Rmd [hunk 1]
```diff
@@ -156,7 +164,9 @@ Alternatively, you can install the development version from GitHub.
 remotes::install_github("r-prof/proffer")
 ```
 
-To use functions `pprof()` and `serve_pprof()`, you need to install [`pprof`](https://github.com/google/pprof). Installing `pprof` is hard, so if you have trouble, please do not hesitate to [open an issue](https://github.com/r-prof/proffer/issues) and ask for help. And if you cannot install `pprof`, then [`profvis`](https://rstudio.github.io/profvis/) is an excellent alternative.
+To use functions `pprof()` and `serve_pprof()`, [`pprof`](https://github.com/google/pprof) needs to be installed. Installing `pprof` is hard, so if you have trouble, please do not hesitate to [open an issue](https://github.com/r-prof/proffer/issues) and ask for help. And if you cannot install `pprof`, then [`profvis`](https://rstudio.github.io/profvis/) is an excellent alternative.
```
Comment (krlmlr) [inline README.Rmd]:
```suggestion
To use functions `pprof()` and `serve_pprof()`, [`pprof`](https://github.com/google/pprof) needs to be installed. This requires a working Go installation and Graphviz. If you have trouble, please do not hesitate to [open an issue](https://github.com/r-prof/proffer/issues) and ask for help. The [`profvis`](https://rstudio.github.io/profvis/) package is an excellent alternative with fewer dependencies.
```

**Review by krlmlr — APPROVED** [review]
Thanks.

We can simplify the Windows installation a bit if we look at default locations for the executables if they're not on the `$PATH`. This seems possible for Go and doesn't look too difficult for Graphviz. Also, we could provide an installation helper that installs {RProtoBuf} and directs the user to installers for Windows, or tells which packages to install on Linux and macOS.

I'll take a stab.

File: README.Rmd [hunk 2]
```diff
@@ -156,7 +164,9 @@ Alternatively, you can install the development version from GitHub.
...
+
+As you follow the installation instructions below, you can run `test_pprof()`, `assert_pprof()`, or `pprof_path()` at any time to see if `proffer` can find and use `pprof`. If these functions succeed early, you are already done.
 
 1. Install the [`RProtoBuf`](https://github.com/eddelbuettel/rprotobuf) package. On Linux, you also need to install the supporting protocol buffer libraries, e.g. `sudo apt-get install protobuf-compiler libprotobuf-dev libprotoc-dev` on Ubuntu.
```
Comment (krlmlr) [inline README.Rmd]: {RProtoBuf} is a strong dependency now. Any problems that have occurred here will already have occurred when installing the package.

```suggestion
```

**Pull request merged**